### ISSUE TITLE
array-like elements have default arguments which violate genericity

### ISSUE THREAD

**Issue by damienwebdev** [faker-js/faker#219]
### Describe the bug

I was reviewing https://github.com/faker-js/faker/pull/189/files and I noticed that `randomize` supports generic results, but there's an issue with the arguments having a default.

Consider:

```ts
const a = randomize<number>();
```

This would return a string, which violates the type.

### Reproduction

[TypeScript Playground Link](https://www.typescriptlang.org/play?ssl=7&ssc=16&pln=1&pc=1#code/GYVwdgxgLglg9mABAJwIZgCZwLYwF4CmAPACqIC8iAzlMjGAOYB8AFKsmgJ4BcKBqWMABtOiEgG0AuhUTiA5KjkAaRHIBGy1RDnTUVROADWYOAHckesVICUvMgG8AsAChEfKCGQWOqTuIAMkgDcLgC+Li4QCDSIqDJomDj4xGAg2GoEyKzWIc6R0XBCBAB0QnAMbDlAA)

**Comment by poyea** [comment]
It seems the violation happens when the argument is undefined / empty. Are we trying to achieve
```typescript
const a: number = randomize<number>();
console.log(a); // 1

const b = randomize();
console.log(b); // "a"
```
or else we may have to think of other ways

**Comment by Shinigami92** [comment]
> It seems the violation happens when the argument is undefined / empty. Are we trying to achieve
> 
> ```ts
> const a: number = randomize<number>();
> console.log(a); // 1
> 
> const b = randomize();
> console.log(b); // "a"
> ```
> 
> or else we may have to think of other ways

Yeah, but sadly this is not possible due to TS is only at compile time.
So the generic cannot change runtime behavior :shrug: 
So we may just remove the default value for the parameter and then declare it as breaking change

**Comment by poyea** [comment]
> > It seems the violation happens when the argument is undefined / empty. Are we trying to achieve
> > ```ts
> > const a: number = randomize<number>();
> > console.log(a); // 1
> > 
> > const b = randomize();
> > console.log(b); // "a"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or else we may have to think of other ways
> 
> Yeah, but sadly this is not possible due to TS is only at compile time. So the generic cannot change runtime behavior 🤷 So we may just remove the default value for the parameter and then declare it as breaking change

Yes if we've to change it. Type specialization in TS is not possible.

**Comment by Shinigami92** [comment]
This function is marked as deprecated (for removal in v7) and we already want to get rid of the default `['a', 'b', 'c']` (#589)
So I'm not sure if it's worth working on this issue specifically

**Comment by ST-DDT** [comment]
Isn't #589 the fix for this issue?

**Comment by xDivisionByZerox** [comment]
Blocked by #893

**Comment by Shinigami92** [comment]
**Team Decision**

Can be implemented without #893

We want this for v8.0
